rinex/src/lib.rs

```diff
@@ -1883,326 +1262,1322 @@ impl Rinex {
+                prev_epoch = Some(epoch);
+                if dt > sample_rate {
+                    Some((epoch, dt))
+                } else {
+                    None
+                }
+            } else {
+                prev_epoch = Some(epoch); // first epoch
+                None
+            }
+        }))
+    }
+}
+
+/*
+ * Methods that return an Iterator exclusively.
+ * These methods are used to browse data easily and efficiently.
+ * It includes Format dependent extraction methods : one per format.
+ */
+use itertools::Itertools; // .unique()
+
+impl Rinex {
+    pub fn epoch(&self) -> Box<dyn Iterator<Item = Epoch> + '_> {
+        if let Some(r) = self.record.as_obs() {
+            Box::new(r.iter().map(|((k, _), _)| *k))
+        } else if let Some(r) = self.record.as_nav() {
+            Box::new(r.iter().map(|(k, _)| *k))
+        } else if let Some(r) = self.record.as_meteo() {
+            Box::new(r.iter().map(|(k, _)| *k))
+        } else if let Some(r) = self.record.as_clock() {
+            Box::new(r.iter().map(|(k, _)| *k))
+        } else if let Some(r) = self.record.as_ionex() {
+            Box::new(r.iter().map(|(k, _)| *k))
+        } else {
+            panic!(
+                "cannot get an epoch iterator for \"{:?}\" RINEX",
+                self.header.rinex_type
+            );
+        }
+    }
+
+    /// Returns a unique [`Sv`] iterator, to navigate
+    /// all Satellite Vehicles encountered and identified.
+    /// This will panic if invoked on ATX, Meteo or IONEX records.
+    /// In case of Clock RINEX, the returns the list of vehicles
+    /// used as reference.
+    /// ```
+    /// use rinex::prelude::*;
+    /// use rinex::sv; // sv!
+    /// use std::str::FromStr; // sv!
+    ///
+    /// let rnx = Rinex::from_file("../test_resources/OBS/V2/aopr0010.17o")
+    ///     .unwrap();
+    /// let mut vehicles : Vec<_> = rnx.sv().collect(); // to run comparison
+    /// vehicles.sort(); // to run comparison
+    ///
+    /// assert_eq!(vehicles, vec![
+    ///     sv!("G01"), sv!("G03"), sv!("G06"),
+    ///     sv!("G07"), sv!("G08"), sv!("G09"),
+    ///     sv!("G11"), sv!("G14"), sv!("G16"),
+    ///     sv!("G17"), sv!("G19"), sv!("G22"),
+    ///     sv!("G23"), sv!("G26"), sv!("G27"),
+    ///     sv!("G28"), sv!("G30"), sv!("G31"),
+    ///     sv!("G32")]);
+    /// ```
+    pub fn sv(&self) -> Box<dyn Iterator<Item = Sv> + '_> {
+        if let Some(record) = self.record.as_obs() {
+            Box::new(
+                // grab all vehicles identified through all Epochs
+                // and fold them into a unique list
+                record
+                    .into_iter()
+                    .map(|((_, _), (_clk, entries))| {
+                        let sv: Vec<Sv> = entries.keys().cloned().collect();
+                        sv
+                    })
+                    .fold(vec![], |mut list, new_items| {
+                        for new in new_items {
```

Comment:
You have these quadratic loops in a couple of places, not sure if performance is a concern here.